mlops-multi-account-cdk/mlops-sm-project-template/seed_code/deploy_app/deploy_endpoint/batch_inference_pipeline.py

```diff
@@ -0,0 +1,346 @@
+            base_job_name=f"{self.pipeline_name}/{self.base_job_prefix}/{step_name}",
+            command=["python3"],
+            sagemaker_session=self.pipeline_session,
+            role=self.role,
+            output_kms_key=self.bucket_kms_id,
+        )
+        BASE_DIR = os.path.dirname(os.path.realpath(__file__))
+
+        output_path = self.get_output_path(step_name)
+        step_args = script_processor.run(
+            # code = f"{BASE_DIR}/source_scripts/preprocessing/main.py",
+            code = f"s3://{self.default_bucket}/{self.pipeline_name}/{self.base_job_prefix}/source-scripts/{step_name}/main.py",
+            outputs=[ProcessingOutput(output_name="transform", source="/opt/ml/processing/output", destination=output_path)],
+            arguments=["--input-data", self.input_data]
+        )
+        step_process = ProcessingStep(
+            name=step_name,
+            step_args=step_args,
+            cache_config=CacheConfig(enable_caching=True, expire_after="T2H")
+        )
+
+        logger.info('Processing step created')
+        return step_process
+    
+
+    def get_transform_step(self, step_process)->TransformStep:
+        ''' 
+        create a transform step to be used in SageMaker pipeline. The transform step will use the model to
+        transform the test dataset
+        
+        Args:
+            step_process: transform step depends on the process step
+            step_create_model: transform step depends on the create model step
+        Returns:
+            TransformStep
+        '''
+        step_name = "transform"
+        output_path = self.get_output_path(step_name)
+
+        transform = Transformer(
+            model_name=self.model_name,
+            instance_count=1,
+            instance_type=self.inference_instance_type,
+            output_path=output_path,
+            base_transform_job_name=f"{self.pipeline_name}/{step_name}",
+            max_payload=10,
+            accept='text/csv'
+        )
+
+        model_transform_step = TransformStep(
+            name=step_name,
+            transformer=transform,
+            inputs=TransformInput(
+                data=step_process.properties.ProcessingOutputConfig.Outputs["transform"].S3Output.S3Uri, 
+                content_type='text/csv'
+            )
+        )
+        logger.info('Transform step created')
+        return model_transform_step
+
+    def get_data_quality_step(self, step_process)->ClarifyCheckStep:
+        ''' 
+        create a data quality step to be used in SageMaker pipeline. The data quality step will compare 
+        data distribution with the baseline, which is generated during training
+
+        Args:
+            step_process: data quality step depends on the process step
+
+        Returns:    
+            ClarifyCheckStep
+        
+        '''
+        from deploy_endpoint.get_approved_package import get_approved_package_desc
+        spec = get_approved_package_desc()
```

Comment:
`get_approved_package_desc()` function is not implemented in the _get_approved_package.py_ file